examples/data_collect/main.cpp

```diff
@@ -275,7 +280,10 @@ int main(int argc, char *argv[]) {
     System system;
     std::vector<std::shared_ptr<Camera>> cameras;
 
-    if (!ip.empty()) {
+    if (!ip.empty() && useNetLinkTest) {
+        ip = "ip:" + ip + ":netlinktest";
+        system.getCameraList(cameras, ip);
```

Comment:
you can call this once, outside of if-else statement

Reply:
Refactored the code to meet both requirements